sympy/matrices/dense.py

```diff
@@ -880,7 +880,10 @@ def row_del(self, i):
         row
         col_del
         """
-        self._mat = self._mat[:i*self.cols] + self._mat[(i + 1)*self.cols:]
+        if i >= -self.rows and i < self.rows:
+            del self._mat[i*self.cols:(i + 1)*self.cols]
+        else:
+            raise IndexError("Index out of range: 'i = %s', valid -%s <= i < %s" % (i, self.rows, self.rows))
```

Comment:
@toolforger Also I think it is not exactly a construction of new matrix with assignment to self._mat that is taking place, with self._mat it was just changing the elements of the matrix by assigning a new list to the elements of matrix. Using del is about 10 times faster in some cases.